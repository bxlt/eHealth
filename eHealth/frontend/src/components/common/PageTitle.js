import React from 'react'
import PropTypes from 'prop-types'

const PageTitle = props => {
  return (
    <div className="page-title">
      {props.optionalTitle ? (
        <div>
          <div className="page-title-optional-main">{props.title}</div>
          <div className="page-title-optional-title">{props.optionalTitle}</div>
        </div>
      ) : (
        <div className="page-title-main">{props.title}</div>
      )}
    </div>
  )
}

PageTitle.propTypes = {
  title: PropTypes.string,
  optionalTitle: PropTypes.string
}

export default PageTitle
