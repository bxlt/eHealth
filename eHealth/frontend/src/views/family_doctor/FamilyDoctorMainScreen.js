import React from 'react'
import PageTitle from '../../components/common/PageTitle'
import MenuCard from '../../components/common/MenuCard'
import FormToBeProcessedSvg from '../../components/clinician/main_screen/FormToBeProcessedSvg'
import CompletedFormSvg from '../../components/clinician/main_screen/CompletedFormSvg'

const FamilyDoctorMainScreen = () => {
  return (
    <div>
      <div>
        <PageTitle title="Family Doctor" />
      </div>
      <div className="card-container">
        <MenuCard
          title="Request Forms"
          svg={FormToBeProcessedSvg}
          route="/family_doctor/patients"
        />
        <MenuCard
          title="Completed Forms"
          svg={CompletedFormSvg}
          route="/family_doctor/completed_forms"
        />
      </div>
    </div>
  )
}

export default FamilyDoctorMainScreen
