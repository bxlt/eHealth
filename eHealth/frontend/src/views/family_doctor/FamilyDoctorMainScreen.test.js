import React from 'react'
import { shallow, mount, render } from "enzyme";
import Enzyme from "enzyme";
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() });

import FamilyDoctorMainScreen from './FamilyDoctorMainScreen'
import PageTitle from './../../components/common/PageTitle';


const loadProps = {
    title: "Family Doctor",
}


describe(`Test <UploaderScreen />`, () => {
    const warpper = shallow(<FamilyDoctorMainScreen />)
    const PageTitleShallowApp = shallow(<PageTitle  {...loadProps} />)
    const PageTitleApp = render(<PageTitle  {...loadProps} />)

    it('1. FamilyDoctorMainScreen App Has been loaded', () => {
        expect(warpper).toMatchSnapshot();
    })

    it('2. PageTitle App Has been loaded', () => {
        expect(PageTitleShallowApp).toMatchSnapshot();
    })

    test('2.1 PageTitle App Has props\'s title\'s Family Doctor', () => {
        var ttext = PageTitleApp.find('.page-title-main').text()
        expect(ttext).toEqual('Family Doctor')
    })

})


