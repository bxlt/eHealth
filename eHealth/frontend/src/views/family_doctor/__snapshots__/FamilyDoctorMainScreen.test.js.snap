// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <UploaderScreen /> 1. FamilyDoctorMainScreen App Has been loaded 1`] = `ShallowWrapper {}`;

exports[`Test <UploaderScreen /> 2. PageTitle App Has been loaded 1`] = `ShallowWrapper {}`;
